### PR TITLE
Add ephemeral toggle to workspace creation page

### PR DESCRIPTION
**Pull request by amisevsk**
### What does this PR do?
Adds an ephemeral mode toggle to the workspace creation page:

eph-toggle

As usual for dashboard changes, the diffs are deceptive due to the [build process](https://github.com/redhat-developer/rh-che/blob/master/assembly/fabric8-ide-dashboard-war/README.md). The files are copied directly from the upstream `6.18.2` release, and modifications are delimited via comments.

### What issues does this PR fix or reference?
#1263 but only for downstream rhche. Upstream dashboard currently does not support configuration via e.g. properties.

### How have you tested this PR?
Locally on minishift.

### CODE REVIEW

File: assembly/fabric8-ide-dashboard-war/src/app/workspaces/create-workspace/create-workspace.controller.ts [hunk 1]
```diff
@@ -0,0 +1,447 @@
...
+   * Property for displaying or hidding the plugins list.
+   */
+  private displayPlugins: boolean;
+
+  /** Begin rhche specific changes */
+  /**
+   * Downstream property for ephemeral mode toggle
+   */
+  private isEphemeralMode: boolean;
+  /** End rhche specific changes */
```
Comment (amisevsk) [inline assembly/fabric8-ide-dashboard-war/src/app/workspaces/create-workspace/create-workspace.controller.ts]: Lines 128-133 changed from upstream.

File: assembly/fabric8-ide-dashboard-war/src/app/workspaces/create-workspace/create-workspace.controller.ts [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+   * Track the changes in ephemeral mode input.
+   */
+  onEphemeralModeChange(): void {
+    if (this.isEphemeralMode) {
+      this.workspaceConfig.attributes.persistVolumes = 'false';
+    } else {
+      delete this.workspaceConfig.attributes.persistVolumes;
+    }
+  }
+  /** End rhche specific changes */
```
Comment (amisevsk) [inline assembly/fabric8-ide-dashboard-war/src/app/workspaces/create-workspace/create-workspace.controller.ts]: Lines 207-218 changed from upstream

File: assembly/fabric8-ide-dashboard-war/src/app/workspaces/create-workspace/create-workspace.html [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+    <!-- Ephemeral mode - rh-che specific -->
+    <che-label-container che-label-name="Ephemeral mode" che-label-description="In ephemeral mode workspace has no PVC attached.">
+      <md-switch class="ephemeral-mode-toggle"
+        ng-model="createWorkspaceController.isEphemeralMode"
+        ng-change="createWorkspaceController.onEphemeralModeChange()"
+        aria-label="ephemeral-mode">
+      </md-switch>
+    </che-label-container>
+    <!-- Ephemeral mode - rh-che specific -->
```
Comment (amisevsk) [inline assembly/fabric8-ide-dashboard-war/src/app/workspaces/create-workspace/create-workspace.html]: Add toggle (lines 84 - 92)

**Comment by amisevsk** [comment]
I tried to delimit the actual changes from upstream to ease review. If anything it's making it more clear that we should look for a better way of customizing the dashboard than our current approach.

**Comment by amisevsk** [comment]
[test]

**Comment by ibuziuk** [comment]
@amisevsk am I correct that ephemeral toggle will be disabled by default? (it should be enabled on osio)

**Comment by amisevsk** [comment]
@ibuziuk Yeah the default setting is off (although that's easily changeable)

**Comment by amisevsk** [comment]
Should be `true` by default now. 

One thing to note: the way ephemeral Che 7 workspaces work (due to plugin broker) means that workspace startup is now actually *longer* than non-ephemeral due to having to run the broker twice.

**Comment by ibuziuk** [comment]
@amisevsk sounds good, was confused by screenshot in the PR description https://user-images.githubusercontent.com/16168279/53780310-d12d3a80-3ed1-11e9-82b1-4f0525d84829.png

**Comment by ibuziuk** [comment]
[test]

**Pull request merged**